### PR TITLE
tasksh 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by opie4624**

### CODE REVIEW

File: Library/Formula/tasksh.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/tasksh", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/tasksh.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (opie4624) [inline Library/Formula/tasksh.rb]: Since this is a directly interactive tool, I'm not sure if any other good test is possible...

In theory, as a test I could write out a tmp file of commands, then run them as a functional test, however I'd worry about that negatively affecting a user's data that may already be present. Thoughts? Does the tmp dir that's created for test reset $HOME for the environment?

Comment (mikemcquaid) [inline Library/Formula/tasksh.rb]: > Does the tmp dir that's created for test reset $HOME for the environment?

Yep, `$HOME` is inside the tmpdir.

**Comment by opie4624** [comment]
Ok, that should be a functional test. And I've checked that it does not stomp on any user data.

File: Library/Formula/tasksh.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    system "echo", "add", "Test Task", "|", "#{bin}/tasksh"
+    system "echo", "1", "del\nyes", "|", "#{bin}/tasksh"
```
Comment (mikemcquaid) [inline Library/Formula/tasksh.rb]: Sorry, this doesn't do what you think it does as it'll just echo the `|` rather than using a pipe. Instead use the `pipe_output` function and `assert_match` on the output.

File: Library/Formula/tasksh.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/tasksh.rb]: You can delete the test above now.

**Comment by tdsmith** [comment]
ping!

**Comment by opie4624** [comment]
Ok, I tried a bunch of other alternatives to come up with a functional test, but I can't come up with anything that actually works properly from the recipe.

The man page states you can pass in a Commands File, but that doesn't work. Echoing commands to a pipe works on the commandline, but it hangs within brew test. (I think it's waiting for a confirmation or exit, since it think's it's interactive.)

Bottom line: I think we'll have to be happy with `--version` for the time being.

**Comment by tdsmith** [comment]
Something like `pipe_output("#{bin}/tasksh", "add Test Task")` fails?

**Comment by opie4624** [comment]
Ah, `pipe_output` will work, but first you have to generate an rcfile. I added the minimum required rc file, and it will now actually test basic functionality. I've left out deleting a task, as that requires interactive confirmation. Successfully adding a task seems like it would adequately surface failures.

**Comment by opie4624** [comment]
Anything else I should do/add?

File: Library/Formula/tasksh.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/tasksh", "--version"
+    rcfile = open(".taskrc", "w")
+    rcfile.write("data.location=~/.task\n")
+    rcfile.close
```
Comment (xu-cheng) [inline Library/Formula/tasksh.rb]:
```ruby
(testpath/".taskrc").write "data.location=#{testpath}/.task\n"
```

**Comment by gusennan** [comment]
ping

**Comment by mikemcquaid** [comment]
@gusennan Not sure what or who you're pinging here.

**Comment by gusennan** [comment]
Since this PR didn't have any activity on it for a while, I copied it to #39278 and made the changes that @xu-cheng requested.  This PR could be closed.
